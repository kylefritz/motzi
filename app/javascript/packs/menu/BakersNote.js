import React from "react";
import { Converter } from "showdown";

export default function ({ bakersNote }) {
  const converter = new Converter({
    encodeEmails: false, // adds uncessary obfuscation to email addresses
    excludeTrailingPunctuationFromURLs: true,
    headerLevelStart: 3,
    simpleLineBreaks: true,
    simplifiedAutoLink: true,
  });
  const noteHtml = converter.makeHtml(bakersNote);
  return (
    <div
      className="bakers-note"
      dangerouslySetInnerHTML={{ __html: noteHtml }}
    />
  );
}
