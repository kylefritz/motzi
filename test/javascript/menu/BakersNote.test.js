import React from "react";
require("../configure_enzyme");
import { shallow } from "enzyme";

import BakersNote from "menu/BakersNote";

test("Menu snapshot", () => {
  const markdown = `# should be h3
  visit www.motzibakery.com.
  email motzi@gmail.com
  `;
  const wrapper = shallow(<BakersNote note={markdown} />);
  expect(wrapper).toMatchSnapshot();
});
