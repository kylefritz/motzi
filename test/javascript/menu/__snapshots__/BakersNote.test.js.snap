// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu snapshot 1`] = `
<div
  className="bakers-note"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<h3 id=\\"shouldbeh3\\">should be h3</h3>
<p>visit <a href=\\"http://www.motzibakery.com\\">www.motzibakery.com</a>.<br />
  email <a href=\\"mailto:motzi@gmail.com\\">motzi@gmail.com</a></p>",
    }
  }
/>
`;
